packages/labs/esbuild/esbuild.bzl

```diff
@@ -0,0 +1,332 @@
+    if _is_supported(f):
+        return f
+
+    no_ext = _strip_ext(f)
+    for i in inputs:
+        if _is_supported(i):
+            if _strip_ext(i) == no_ext:
+                return i
+    fail("Could not find corresponding javascript entry point for %s. Add the %s.js to your deps." % (f.path, no_ext))
+
+def _filter_supported_files(files):
+    return [f for f in files if _is_supported(f)]
+
+def _resolve_entry_points(file, files, inputs):
+    if file != None:
+        files = [file]
+
+    return [_resolve_input(f, inputs) for f in files]
+
+def _es_build_impl(ctx):
+    # For each dep, JSEcmaScriptModuleInfo is used if found, then JSModuleInfo and finally
+    # the DefaultInfo files are used if the former providers are not found.
+    deps_depsets = []
+    for dep in ctx.attr.deps:
+        if JSEcmaScriptModuleInfo in dep:
+            deps_depsets.append(dep[JSEcmaScriptModuleInfo].sources)
+        elif JSModuleInfo in dep:
+            deps_depsets.append(dep[JSModuleInfo].sources)
+        elif hasattr(dep, "files"):
+            deps_depsets.append(dep.files)
+
+        if NpmPackageInfo in dep:
+            deps_depsets.append(dep[NpmPackageInfo].sources)
+
+    deps_inputs = depset(transitive = deps_depsets).to_list()
+    inputs = _filter_supported_files(ctx.files.entry_point) + ctx.files.srcs + deps_inputs
+
+    if ctx.file.tsconfig:
+        inputs.append(ctx.file.tsconfig)
+        if TsConfigInfo in ctx.attr.tsconfig:
+            inputs.extend(ctx.attr.tsconfig[TsConfigInfo].deps)
+
+    outputs = [getattr(ctx.outputs, o) for o in dir(ctx.outputs)]
+    if ctx.attr.outdir:
+        outputs.append(ctx.actions.declare_directory(ctx.label.name))
+
+    entry_points = _resolve_entry_points(ctx.file.entry_point, ctx.files.entry_points, inputs)
+
+    args = ctx.actions.args()
+    args.add_all([f.path for f in entry_points])
+    args.add("--bundle")
+    if ctx.attr.minify:
+        args.add("--minify")
+    if ctx.attr.splitting:
+        args.add("--splitting")
+    if ctx.attr.format:
+        args.add_joined(["--format", ctx.attr.format], join_with = "=")
+    if ctx.attr.metafile:
+        metafile = ctx.actions.declare_file("%s_esbuild_metafile.json" % ctx.attr.name)
+        outputs.append(metafile)
+        args.add_joined(["--metafile", metafile.path], join_with = "=")
+
+    if ctx.attr.sourcemap != "default":
+        args.add_joined(["--sourcemap", ctx.attr.sourcemap], join_with = "=")
+    elif ctx.attr.sourcemap == "default":
+        args.add("--sourcemap")
+
+    if ctx.attr.outdir:
+        args.add_joined(["--outdir", outputs[0].path], join_with = "=")
+    else:
+        args.add_joined(["--outfile", outputs[0].path], join_with = "=")
+
+    args.add_joined(["--platform", ctx.attr.platform], join_with = "=")
+    args.add_all(ctx.attr.loader, format_each = "--loader=%s")
+    args.add_all(ctx.attr.define, format_each = "--define=%s")
+    args.add_joined(["--target", "esnext"], join_with = "=")
```

Comment:
Should the target be configurable? If not could you leave a comment saying why not?